src/lib/scheduler/abstract_task.cpp

```diff
@@ -23,11 +23,14 @@ NodeID AbstractTask::node_id() const { return _node_id; }
 
 bool AbstractTask::is_ready() const { return _pending_predecessors == 0; }
 
-bool AbstractTask::is_done() const { return _done; }
+bool AbstractTask::is_done() const { return _state == TaskState::Done; }
 
 bool AbstractTask::is_stealable() const { return _stealable; }
 
-bool AbstractTask::is_scheduled() const { return _is_scheduled; }
+bool AbstractTask::is_scheduled() const {
+  DebugAssert(static_cast<std::uint8_t>(TaskState::Created) == 0, "TaskState::Created is not equal to 0");
```

Comment:
(1) Why do we need this check? This somewhat looks like you are asserting the behavior of the language itself.
(2) Can't it be a `static_assert`?

Reply:
I use the `uint8_t` values of `OperatorState` for easy comparison. 
When someone changes `OperatorState::Created` to `1` this function will be broken.

Reply:
I guess it doesn't hurt, but it should be a static_assert.